packages/@glimmer/test-helpers/lib/abstract-test-case.ts

```diff
@@ -163,12 +163,38 @@ export abstract class AbstractRenderTest {
     return invocation;
   }
 
-  private buildArgs(args: Object, sigil = ""): string {
-    return `${Object.keys(args).map(arg => `${sigil}${arg}=${args[arg]}`).join(" ")}`;
+  private buildArgs(args: Object): string {
+    let { testType } = this;
+    let sigil = "";
+    let needsCurlies = false;
+
+    if (testType === "Glimmer" || testType === "Basic") {
+      sigil = "@";
+      needsCurlies = true;
+    }
+
+    return `${Object.keys(args)
+      .map(arg => {
+        let rightSide: string;
+
+        if (needsCurlies) {
+          let isString = arg[0] === "'" || arg[0] === '"';
+          if (isString) {
+            rightSide = `${args[arg]}`;
+          } else {
+            rightSide = `{{${args[arg]}}}`;
+          }
+        } else {
+          rightSide = `${args[arg]}`;
+        }
+
+        return `${sigil}${arg}=${rightSide}`;
+      })
+      .join(" ")}`;
   }
 
   private buildBlockParams(blockParams: string[]): string {
-    return `${blockParams.length > 0 ? `as |${blockParams.join(" ")}|` : ""}`;
+    return `${blockParams.length > 0 ? ` as |${blockParams.join(" ")}|` : ""}`;
```

Comment:
Do you think it's worth defining `const SPACE = " ";`. You use it quite a few times in this file (though I realize this is just a test infra file)?